app/controllers/profile_controller.rb

```diff
@@ -0,0 +1,12 @@
+class ProfileController < ApplicationController
+
+  # GET /profile
+  def index
+    begin
+      @user = User.find(session[:user_id])
+    rescue StandardError => e
+      redirect_to '/unauthorized'
+    end
+  end
```

Comment:
I think you can simplify this code to check whether the user id from the session is nil and if so, redirect to the unauthorized page. You should also update the status to 403.

Reply:
I need to get the user to render the page anyways